crates/data_components/src/debezium/arrow.rs

```diff
@@ -427,6 +427,84 @@ fn convert_string_to_decimal(field_value: &str) -> Result<i128> {
     Ok(decimal_i128)
 }
 
+#[inline]
+fn pow10_i128(exp: i8) -> Option<i128> {
+    let mut acc: i128 = 1;
+    for _ in 0..exp {
+        acc = acc.checked_mul(10)?;
+    }
+    Some(acc)
+}
+
+fn rescale_i128(unscaled: i128, src_scale: i8, dst_scale: i8) -> Result<i128> {
+    match src_scale.cmp(&dst_scale) {
+        Equal => Ok(unscaled),
+        Less => {
+            let diff = dst_scale - src_scale;
+            let mul = pow10_i128(diff).context(InvalidDecimalJsonSnafu {
+                reason: format!("overflow computing 10^{diff}"),
```

Comment:
nit: instead of having a single error with multiple static reasons, just create multiple error variants.

Reply:
Done